### DISCUSSION TITLE
uc_click does not reconnect to the same tab the click occurred in.

### DISCUSSION THREAD

**Discussion by Dylgod** [seleniumbase/SeleniumBase#2791] [Q&A] [answered]
Below is a test to show this behavior. This issue happened to me when i was trying to fill out a form on a second tab using uc mode.
Is this the intended behavior or is there a more convenient way to use this? Ideally I would like to be able to use uc_click at will on a page without the need to switch tabs after every uc_click.

```Python
from seleniumbase import Driver
from selenium.webdriver.common.by import By

driver = Driver(uc=True)
try:
    driver.get("https://www.freecodecamp.org/news/how-to-use-html-to-open-link-in-new-tab/")
    print("Start", '\n', driver.current_url, '\n')
    new_tab_link = driver.find_element(By.XPATH, '//a[@target="_blank" and contains(text(), "freeCodeCamp")]')
    new_tab_link.uc_click()

    anchor_handle = driver.current_window_handle
    tabs = driver.window_handles
    for handle in tabs:
        driver.switch_to.window(handle)
        if driver.current_window_handle != anchor_handle:
            break

    # confirming driver is on the second tab
    print("Switched to second tab", '\n', driver.current_url, '\n')

    getstarted_btn = driver.find_element(By.XPATH, '//a[@href="https://api.freecodecamp.org/signin"]/span[contains(text(), "Get started (it\'s free)")]')
    getstarted_btn.uc_click()

    # confirming driver reconnected to first tab
    print("After uc_click in second tab", '\n', driver.current_url)
finally:
    driver.quit()
```

P.S. I sure you had a blast at the Star Trek convention! I hope you wore your engineering uniform 🖖🏻

**Comment by mdmintz** [comment, marked as answer]
For stealth, one component of `uc_click` is reconnecting chromedriver to Chrome. This can change tab-order if multiple tabs are open because variables are lost during the disconnect (controlled by raw Selenium, not SeleniumBase).

To avoid tab-reordering, use a single tab for UC Mode scripts. More info: https://github.com/seleniumbase/SeleniumBase/issues/2328#issuecomment-1834021680

----

The Star Trek convention was great! Already posted photos with Jonathan Frakes and LeVar Burton: https://github.com/seleniumbase/SeleniumBase/issues/2785#issuecomment-2119528065

Here's a few more photos from the set tour:

<img width="300" alt="Trekonderoga Star Trek convention" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/533787cf-8b27-4eb3-b20d-c9c86635958d">

<img width="447" alt="Trekonderoga Star Trek convention" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/93a2dda1-a1e3-45e5-9e18-9ec5dfb91988">

I'll post videos with Jonathan Frakes and LeVar Burton later on YouTube. I spoke with them and Walter Koenig.
